src/main/scala/com/microsoft/hyperspace/telemetry/HyperspaceEventLogging.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.microsoft.hyperspace.telemetry
+
+import scala.util.{Success, Try}
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.util.hyperspace.Utils
+
+import com.microsoft.hyperspace.HyperspaceException
+import com.microsoft.hyperspace.telemetry.Constants._
+
+/**
+ * Hyperspace event logging interface. Extend this to enable emitting hyperspace events.
+ */
+trait HyperspaceEventLogging {
+  def logEvent(event: HyperspaceEvent): Unit = EventLogger.logger.logEvent(event)
+}
+
+/**
+ * Event Logger interface. Concrete implementations of this class implement event handling logic
+ * for hyperspace events.
+ */
+trait EventLogger {
+  def logEvent(event: HyperspaceEvent): Unit
+}
+
+object EventLogger extends Logging {
+  // Singleton logger instance for event logging.
+  private[telemetry] lazy val logger: EventLogger = SparkSession.getActiveSession
```

Comment:
Did you want to scope this at the `object` level instead? `private[telemetry] EventLogger`. Not sure of your intention here.

Reply:
Oh I made a slight change. Instead of using a getLogger public method, I directly used the logger object in the `HyperspaceLogging class. So I thought of making it private[telemetry]

Reply:
I can revert it to
```
private lazy val logger: EventLogger = ....

def getLogger: EventLogger = logger
```

Reply:
Oh this is fine. I was just curious if we should apply the scoping at the object level.

Reply:
sorry I misread the comment. Will scope it to object level